packit_service/worker/checker/run_condition.py

```diff
@@ -0,0 +1,204 @@
+            task_name=task_name,
+        )
+        self._handler_kls = None
+        self._working_dir: Optional[Path] = None
+        self._command_handler: Optional[CommandHandler] = None
+        self._actions_handler: Optional[ActionsHandler] = None
+
+    @property
+    def handler_kls(self):
+        if self._handler_kls is None:
+            logger.debug(f"Command handler: {self.service_config.command_handler}")
+            self._handler_kls = RUN_COMMAND_HANDLER_MAPPING[self.service_config.command_handler]
+        return self._handler_kls
+
+    @property
+    def working_dir(self) -> Optional[Path]:
+        if not self._working_dir:
+            if self.handler_kls == SandcastleCommandHandler:
+                path = (
+                    Path(self.service_config.command_handler_work_dir) / "run-condition-working-dir"
+                )
+                path.mkdir(parents=True, exist_ok=True)
+                self._working_dir = path
+            else:
+                self._working_dir = Path(tempfile.mkdtemp())
+            logger.info(
+                f"Created directory for the run-condition action: {self._working_dir}",
+            )
+        return self._working_dir
+
+    @property
+    def command_handler(self) -> CommandHandler:
+        if self._command_handler is None:
+            self._command_handler = self.handler_kls(
+                config=self.service_config,
+                working_dir=self.working_dir,
+            )
+        return self._command_handler
+
+    @property
+    def actions_handler(self) -> ActionsHandler:
+        if not self._actions_handler:
+            self._actions_handler = ActionsHandler(
+                self.job_config,
+                self.command_handler,
+            )
+        return self._actions_handler
+
+    def common_env(
+        self, version: Optional[str] = None, extra_env: Optional[dict[str, str]] = None
+    ) -> dict[str, str]:
+        env = self.job_config.get_package_names_as_env()
+        if version:
+            env["PACKIT_PROJECT_VERSION"] = version
+        if extra_env:
+            env.update(extra_env)
+        return env
+
+    def pre_check(self) -> bool:
+        project = self.project
+        git_ref = self.data.commit_sha
+        version = None
+
+        if self.data.event_type in (
+            github.pr.Action.event_type(),
+            pagure.pr.Action.event_type(),
+            gitlab.mr.Action.event_type(),
+            github.pr.Comment.event_type(),
+            pagure.pr.Comment.event_type(),
+            gitlab.mr.Comment.event_type(),
+            pagure.pr.Flag.event_type(),
+            github.check.PullRequest.event_type(),
+        ):
+            project = self.project.get_pr(int(self.data.pr_id)).source_project
+        elif self.data.event_type in (anitya.NewHotness.event_type(),):
+            project = self.service_config.get_project(
+                url=self.data.event_dict.get("distgit_project_url")
+            )
```

Comment:
is this actually needed? (isn't it just`self.project` also?)

Reply:
Yes, `self.project` is the upstream project (associated with the issue).

Reply:
from [this](https://github.com/packit/packit-service/blob/main/packit_service/events/anitya/abstract.py#L48-L58) it looks like the switching happens in the event class, or am I missing something?

Reply:
> from [this](https://github.com/packit/packit-service/blob/main/packit_service/events/anitya/abstract.py#L48-L58) it looks like the switching happens in the event class, or am I missing something?

Yes, but that's not propagated to the dict returned from `get_dict()` that is passed to the checker. But I stand corrected, `self.project` is not an upstream project in this case, it is `None`.

Reply:
That makes me think if rewriting the code so that the processing is based on task name rather than event type and associated event objects are reconstructed from event data wouldn't make more sense, it could at least be more readable.

Reply:
might be, I am also still thinking if this couldn't be implemented on handlers as some class methods maybe, without hitting the circular imports for that.

Reply:
I'm not sure the circular import can be avoided, unless the "checker" class itself is defined in the same module as handlers (or handler abstract classes).